### PR TITLE
Add support to ServerNameIndicators for load balancers

### PR DESCRIPTION
**Pull request by mouhsen-ibrahim**

### CODE REVIEW

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 1]
```diff
@@ -55,10 +55,16 @@ type Model struct {
...
 // Struct corresponding to each Model.Listener
 type Listener struct {
-	DisplayName types.String `tfsdk:"display_name"`
-	Port        types.Int64  `tfsdk:"port"`
-	Protocol    types.String `tfsdk:"protocol"`
-	TargetPool  types.String `tfsdk:"target_pool"`
+	DisplayName          types.String          `tfsdk:"display_name"`
+	Port                 types.Int64           `tfsdk:"port"`
+	Protocol             types.String          `tfsdk:"protocol"`
+	ServerNameIndicators []ServerNameIndicator `tfsdk:"serverNameIndicators"`
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer/resource.go:61]: We have a known bug that using a custom type for the resource fields causes an error when providing the values as a variable. So we will replace all of these custom types by `types.Object`. 

We planned on doing it in this iteration, so we can jump in here and replace it for this field also

Comment (hcsa73) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go]: Modified it

File: stackit/internal/services/loadbalancer/loadbalancer_acc_test.go [hunk 2]
```diff
@@ -71,6 +72,11 @@ func configResources(targetPort string) string {
 				display_name = "%s"
 				port         = %s
 				protocol     = "%s"
+				serverNameIndictors = [
```
Comment (vicentepinto98) [inline stackit/internal/services/loadbalancer/loadbalancer_acc_test.go:75]: field name is wrong

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mouhsen-ibrahim) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go]: fixed it

File: stackit/internal/services/loadbalancer/loadbalancer/datasource.go [hunk 3]
```diff
@@ -147,6 +149,18 @@ func (r *loadBalancerDataSource) Schema(_ context.Context, _ datasource.SchemaRe
 							Description: descriptions["protocol"],
 							Computed:    true,
 						},
+						"serverNameIndicators": schema.ListNestedAttribute{
```
[the remark this answers is not in the document]
Comment (mouhsen-ibrahim) [reply stackit/internal/services/loadbalancer/loadbalancer/datasource.go]: fixed it

File: stackit/internal/services/loadbalancer/loadbalancer_acc_test.go (hunk 2 above)
Comment (mouhsen-ibrahim) [reply stackit/internal/services/loadbalancer/loadbalancer_acc_test.go]: fixed it

File: stackit/internal/services/loadbalancer/loadbalancer/resource.go [hunk 4]
```diff
@@ -305,7 +313,19 @@ The example below uses OpenStack to create the network, router, a public IP addr
 								stringplanmodifier.UseStateForUnknown(),
 							},
 							Validators: []validator.String{
-								stringvalidator.OneOf("PROTOCOL_UNSPECIFIED", "PROTOCOL_TCP", "PROTOCOL_UDP", "PROTOCOL_TCP_PROXY"),
+								stringvalidator.OneOf("PROTOCOL_UNSPECIFIED", "PROTOCOL_TCP", "PROTOCOL_UDP", "PROTOCOL_TCP_PROXY", "PROTOCOL_TLS_PASSTHROUGH"),
+							},
+						},
+						"serverNameIndicators": schema.ListNestedAttribute{
```
[the remark this answers is not in the document]
Comment (mouhsen-ibrahim) [reply stackit/internal/services/loadbalancer/loadbalancer/resource.go]: fixed it

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**